### PR TITLE
MyDumper hangs with --stream option if DB is big

### PR DESCRIPTION
**Pull request by midenok**
start_dump() thread first does free_db_table(dbt) then
wait_stream_to_finish(). This is wrong order as
metadata_partial_writer() thread (waited by stream thread) accesses
dbt.

The fix reoders wait_stream_to_finish() so that is done before
free_db_table().

Bug description:

https://jira.mariadb.org/browse/CONTRIB-10
@midenok

### CODE REVIEW

File: src/mydumper_start_dump.c
Comment (davidducos) [inline src/mydumper_start_dump.c:1]: mdfile needs to be close before streaming metadata_filename

Comment (midenok) [reply src/mydumper_start_dump.c:1]: > mdfile needs to be close before streaming metadata_filename

Why?

Comment (midenok) [reply src/mydumper_start_dump.c:1]: @davidducos I can do investigation myself, but if you have some quck knowledge that saves me time, please share!

File: src/mydumper_start_dump.c [hunk 1]
```diff
@@ -1248,26 +1263,11 @@ void start_dump() {
   fclose(mdfile);
```
Comment (davidducos) [inline src/mydumper_start_dump.c:1263]: Here is where mdfile is being close, but you are streaming metadata before closing. 
The stream backup works like this:
- you open the file and save all the info
- you close the file to guarantee that everything is in there
- you send the file to the stream thread, which is going to reopen, send the file through the stdout and close it again
You should move the close before enqueueing into the stream queue if nothing is been writing in mdfile

Comment (midenok) [reply src/mydumper_start_dump.c:1263]: > you close the file to guarantee that everything is in there

Then I guess flushing must be enough. There are chicken and egg problem for doing this after close.

**Comment by midenok** [comment]
Duplicates #1355

**Pull request closed without merging**